### PR TITLE
Fix formatting inconsistencies

### PR DESCRIPTION
**Pull request by mattmcneeney**
The spec had become a bit inconsistent, so this is a sweeping change to tidy it up a bit and make it pretty. Let me know what you think and LGTM if happy, thanks!

### CODE REVIEW

**Comment by angarg12** [comment]
Should we wait for the other spec changes that are about to be accepted to rebase and merge this one?

**Comment by mattmcneeney** [comment]
Yeah good idea, it would probably best to merge imminent ones and then rebase this one before getting LGTMs. I'll bump this in a day or so.

**Comment by duglin** [comment]
SGTM and yeah let's wait for the other PRs to go first since it'll be rebase hell and some of these fixes are part of those PRs too.

**Comment by mattmcneeney** [comment]
Had another run through this in attempt to avoid rebase hell. Let's get any imminent PRs merged so we can sort this one out! 🙏

File: spec.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ A platform marketplace MAY expose services from one or many service brokers, and
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to
```
Comment (duglin) [inline spec.md]: extra space before "MAY".  should just search for double-spaces in the entire doc.  :-)

File: spec.md [hunk 2]
```diff
@@ -85,68 +85,68 @@ Warnings for broker authors:
...
-##### Service Objects #####
+##### Service Objects
 
-|  Response field |  Type | Description  |
-|---|---|---|
-|  name* | string  |  A CLI-friendly name of the service. All lowercase, no spaces. This MUST be globally unique within a platform marketplace. |
-| id*  |  string | An identifier used to correlate this service in future requests to the broker. This MUST be globally unique within a platform marketplace. Using a GUID is RECOMMENDED.  |
-|  description* |  string | A short description of the service.  |
-| tags  |  array-of-strings | Tags provide a flexible mechanism to expose a classification, attribute, or base technology of a service, enabling equivalent services to be swapped out without changes to dependent logic in applications, buildpacks, or other services. E.g. mysql, relational, redis, key-value, caching, messaging, amqp.  |
-|  requires | array-of-strings  | A list of permissions that the user would have to give the service, if they provision it. The only permissions currently supported are <code>syslog\_drain</code>, <code>route\_forwarding</code> and <code>volume\_mount</code>.  |
```
Comment (duglin) [inline spec.md]: very happy to see these "code" blocks being replace with back-ticks!

**Comment by duglin** [comment]
one minor double-space missed but I want this merged ASAP, so
LGTM

**Comment by duglin** [comment]
@mattmcneeney let's do the 80 column wrap in a follow-on PR

**Comment by angarg12** [comment]
LGTM!

**Comment by shalako** [comment]
My expectation is this will break CF docs, as:
- pound signs on both sides of the title are used for auto-generating table of contents
- backticks in tables are recognized for formatting, so html is necessary

This may or may not still be the case.
@mattmcneeney, please check with the docs team if you haven't already. Thanks!

**Comment by vaikas-google** [comment]
LGTM

**Comment by mattmcneeney** [comment]
@duglin thanks for the heads up. I'll check with the CF docs team asap, so let's not merge this one just yet. In the meantime, I'm about to push a fix for the double spaces @duglin

**Comment by duglin** [comment]
I have a feeling that we may run into a lot of problems in the future if we try to continue down this path of our spec being in markdown as well as an html-md-hybrid that the CF doc tooling needs.  Someone once mentioned the name of the tool that CF uses but I can't remember -what is it?  I'd like to explore the option of helping the CF guys out by creating a tool that will take a "normal" github md docs and twiddle it so that their tooling will work - similar to what I did in https://github.com/openservicebrokerapi/servicebroker/pull/168#issuecomment-297045092 . Then we could stick to just github md.

**Comment by avade** [comment]
@duglin https://github.com/pivotal-cf/bookbinder is the tool. @bentarnoff would be the person to speak to, but I don't expect it would be a small piece of work.

**Comment by avade** [comment]
Is this good for another round of LG's?

**Comment by duglin** [comment]
@avade what about @shalako's comment?  Are CF docs ok with this?

**Comment by duglin** [comment]
@bentarnoff @avade I noticed this https://github.com/pivotal-cf/bookbinder#middleman-templating-helpers  I wonder if we could create a helper to tweak the MD into something bookbinder would be ok with?  Or, if not, what does it use to pull down the repo? If its just "git" in the path then we could wrap git with a bash script to twiddle things after the clone.

**Comment by avade** [comment]
Let's wait to hear from @bentarnoff before merging. 

CF docs are the only place the spec is rendered at the moment.

**Comment by mattmcneeney** [comment]
I've checked out the CF docs locally and have been playing around with what we can and can't change as of today. Results as follows:
* Backticks in tables are fine
* Navigation is broken with or without #'s on the other side of titles and doesn't affect ToC (they have been broken since the anchor tags were removed a while ago, and we don't use bookbinder to generate ToC for the API docs)
* Using two stars for bold text (`**Text**`) is not supported inside `<p class="note">`
* Three backticks appears differently to `<pre class="terminal">`

@bentarnoff LMK what you think of this list. If we could fix navigation and, if possible, change a pair of three backticks to render `<pre class="terminal">...content...</pre>` either in CI or through a custom middleman helper then we can get this spec tidied up.

**Comment by mattmcneeney** [comment]
I've submitted a [PR to the bookbinder repo](https://github.com/pivotal-cf/bookbinder/pull/115) that fixes the navigation. We can make do with the `<pre class="terminal"></pre>` for now and hopefully get this merged once the docs PR gets approved.

**Comment by mattmcneeney** [comment]
We've removed all HTML tags now except `<p class="note">` as we don't have a way in markdown yet to do nice things like that. The rest of the spec now is pretty pure though.

**Comment by bentarnoff** [comment]
Let me bring in @animatedmax, who can speak to the `bookbinder` PR.

Per @avade's request, CF docs is now pulling from the `v2.11` branch of this repo. But it doesn't appear to have solved the ToC issue, which you can see is still a problem on our OSS staging site: https://docs-cloudfoundry-staging.cfapps.io/services/api.html

**Comment by animatedmax** [comment]
The [bookbinder PR](https://github.com/pivotal-cf/bookbinder/pull/115) appears to resolve the anchor issue. It adds an extra id to the anchors we manually add to pages, but I believe they have no appreciable impact on these existing pages.

**Comment by mattmcneeney** [comment]
Great, thanks Max. I think this is now ready for a round of LGTMs then!

**Comment by angarg12** [comment]
LGTM!

**Comment by mattmcneeney** [comment]
Three more please 🥇  @avade @duglin @pmorie and co

**Comment by shalako** [comment]
LGTM

**Comment by avade** [comment]
LGTM

**Comment by vaikas-google** [comment]
LGTM

**Pull request merged**